### PR TITLE
PP-6343: Replace Jedis with lettuce-core library

### PR DESCRIPTION
**Pull request by oswaldquek**
Follows the basic usage according to https://github.com/lettuce-io/lettuce-core/wiki/Basic-usage

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java [hunk 1]
```diff
@@ -52,20 +53,14 @@ synchronized void checkRateOf(String accountId, RateLimiterKey key)
 
     synchronized private Long updateAllowance(String key) {
         String derivedKey = getKeyForWindow(key);
+        StatefulRedisConnection<String, String> connection = redisClient.connect();
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java:56]: Do we need to close this connection or does lettuce takes care of it ? Java docs says `connect()` opens new connection to Redis https://lettuce.io/core/release/api/io/lettuce/core/RedisClient.html#connect-- 

Also with Jedis, we get connection pool. Is this something that will be useful ?

**Review by kbottla — COMMENTED** [review]
Looks good to me. Just a comment about managing connection (ie., if it is to be closed explicitly or management comes with lettuce by default) and if a connection pool is useful if we are opening connection for every request

File: src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java (hunk 1 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java:56]: Thanks for catching this out. I think the `StatefulRedisConnection` can be a field in the class. However, given that `updateAllowance` is a synchronized method, I don't think there's any value in using a connection pool here.

Comment (oswaldquek) [reply src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java]: @kbottla I've closed the connection via the (renamed) RedisClientManager now.

**Review by kbottla — COMMENTED** [review]
Looks good to me.  
A minor comment about using wrapper for RedisClient to keep connection management from RedisRateLimiter, but is not blocking..

Would be good to get more eyes on ADR and if we want to plan further steps to make Redis mandatory in all environments.

File: src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java [hunk 2]
```diff
@@ -96,4 +99,9 @@ private String getKeyForWindow(String key) throws OutOfScopeException {
         return key + window;
     }
 
+    public void closeRedisConnection() {
```
[the remark this answers is not in the document]
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java]: Good shout. I've now managed the RedisClient connection from within the RedisClientManager instead.

[the remark this answers is not in the document]
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java]: I felt it was a little over the top to create a wrapper class.

**Review by kbottla — APPROVED** [review]
LGTM!

**Pull request merged**